include/swift/SIL/SILInstructionWorklist.h

```diff
@@ -24,92 +24,127 @@
 ///
 //===----------------------------------------------------------------------===//
 
+#include "swift/Basic/BlotSetVector.h"
 #include "swift/SIL/SILInstruction.h"
 #include "swift/SIL/SILValue.h"
 #include "llvm/ADT/DenseMap.h"
 #include "llvm/ADT/SmallVector.h"
 
 namespace swift {
 
-/// Manages a list of instructions awaiting visitation.
-class SILInstructionWorklist final {
-  llvm::SmallVector<SILInstruction *, 256> worklist;
-  llvm::DenseMap<SILInstruction *, unsigned> worklistMap;
+class SILInstructionWorklistBase {
   StringRef loggingName;
```

Comment:
@nate-chandler @gottesmm why is a `StringRef` stored here? That's illegal. Given the purpose, a `const char *` underneath `NDEBUG` should be fine.

Reply:
We are careful with how we construct it. Specifically, if you look this is only set in the constructor from a const char *. So it /is/ safe.

Reply:
That being said, if you feel this is a code smell, I do not have any objections to storing a const char *. I just prefer to have a StringRef handle to a const char * in general.

Reply:
@atrick It's now a StringLiteral which @gottesmm pointed me at, does that sound okay?

Reply:
@nate-chandler `StringLiteral` is fine with me; pointless really in this context, but it's just a matter of style.

@gottesmm Storing a `StringRef` in something that isn't guaranteed to be transient isn't code smell, it's a perversion of the abstraction. The purpose of `StringRef`, and only reason to use it, is in an API that guarantees only transient use of the value.

Reply:
I disagree. To me that is just one use. For instance consider how we use identifiers in the compiler. Those are referenced via StringRefs in a bunch of places.

That being said, I don't think this is an important thing to argue about = p.